File: video-service/src/main/java/com/service/page/LoginPage.java
package com.service.page;

import com.automation.remarks.kirk.Browser;
import com.automation.remarks.kirk.Page;

/**
 * Created by sepi on 26.08.17.
 */
public class LoginPage extends Page {
  public LoginPage(Browser browser) {
    super(browser);
  }

  @Override
  public String getUrl() {
    return "/";
  }

  public void loginAs(String name, String password) {
    element("#inputEmail3").setValue(name).pressEnter();
    element("#inputPassword3").setValue(password);
    element("#parent > form > div:nth-child(3) > div > button").click();
  }
}

File: src/main/kotlin/pages/LoginPage.kt
package pages

import com.automation.remarks.kirk.Page
import domain.User
import io.qameta.allure.Step

/**
 * Created by sergey on 17.06.17.
 */
class LoginPage : Page() {

    override val url: String?
        get() = "http://localhost:8086"

    val username = element("#inputEmail3")
    val password = element("#inputPassword3")
    val loginBtn = element("#parent > form > div:nth-child(3) > div > button")

    @Step fun loginAs(user: User) {
        username.setVal(user.name)
        password.setVal(user.password)
        loginBtn.click()
    }
}